src/include/compat.h

```diff
@@ -195,24 +216,32 @@ struct cpu_set_t;
     pthread_set_name_np(thread, name);          \
     0; })
 #else
-  /* compiler warning free success noop */
-  #define ceph_pthread_setname(thread, name) ({ \
-    int __i = 0;                              \
-    __i; })
+  #define ceph_pthread_setname(thread, name) \
+    ceph_pthread_setname_noop(thread, name)
```

Comment:
I think this can be simplified:
```suggestion
  #define ceph_pthread_setname ceph_pthread_setname_noop
```